zuul-core/src/test/java/com/netflix/zuul/message/HeadersTest.java

```diff
@@ -202,6 +202,30 @@ public void setNullIsEmtpy_headerName() {
         Truth.assertThat(headers.size()).isEqualTo(1);
     }
 
+    @Test
+    public void setIfValid() {
+        Headers headers = new Headers();
+        headers.add("Via", "duct");
+        headers.add("Cookie", "this=that");
+        headers.add("Cookie", "frizzle=frazzle");
+        headers.add("X-Valid-K1", "abc-xyz");
+        headers.add("X-Valid-K2", "def-xyz");
+        headers.add("X-Valid-K3", "xyz-xyz");
+
+        headers.setIfValid(new HeaderName("cookIe"), null);
+        headers.setIfValid(new HeaderName("X-Valid-K1"), "abc\r\n-xy\r\nz");
+        headers.setIfValid("X-Valid-K2", "abc\r-xy\rz");
+        headers.setIfValid("X-Valid-K3", "abc\n-xy\nz");
+        headers.setIfValid("X-Valid-K4", "abc-abc");
+
+        Truth.assertThat(headers.getAll("CookiE")).isEmpty();
+        Truth.assertThat(headers.getAll("X-Valid-K1")).containsExactly("abc-xyz");
+        Truth.assertThat(headers.getAll("X-Valid-K2")).containsExactly("def-xyz");
+        Truth.assertThat(headers.getAll("X-Valid-K3")).containsExactly("xyz-xyz");
+        Truth.assertThat(headers.getAll("X-Valid-K4")).containsExactly("abc-abc");
+        Truth.assertThat(headers.size()).isEqualTo(5);
+    }
```

Comment:
Do you think that is possible to break this test in multiple small tests/behaviors?

Reply:
Done